email.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+import logging, os
+from typing import Any, Union, List, Dict, Tuple
+from viur.core.config import conf
+from viur.core import db, utils
+from viur.core.utils import projectID
+from viur.core.tasks import callDeferred, QueryIter, PeriodicTask
+import json, base64
+from urllib import request
+
+"""
+	This module implements an email delivery system for ViUR. Emails will be queued so that we don't overwhelm
+	the email service. As the Appengine does not provide an email-api anymore, you'll have to use a 3rd party service
+	to actually deliver the email. A sample implementation for Send in Blue (https://sendinblue.com/) is provided.
+	To enable Send in Blue,	set conf["viur.email.transportFunction"] to transportSendInBlue and add your API-Key to
+	conf["viur.email.sendInBlue.apiKey"]. To send via another service, you'll have to implement a different transport
+	function (and point conf["viur.email.transportFunction"] to that function).
+"""
+
+@PeriodicTask(interval=60*24)
+def cleanOldEmailsFromLog(*args, **kwargs):
+	"""
+		Start the QueryIter DeleteOldEmailsFromLog to remove old, successfully send emails from the queue
+	"""
+	qry = db.Query("viur-emails").filter("isSend =", True) \
+		.filter("creationDate <", utils.utcNow()-conf["viur.email.logRetention"])
+	DeleteOldEmailsFromLog.startIterOnQuery(qry)
+
+class DeleteOldEmailsFromLog(QueryIter):
+	"""
+		Simple Query-Iter to delete all entities encountered
+	"""
+	@classmethod
+	def handleEntry(cls, entry, customData):
+		db.Delete(entry.key)
+
+
+@callDeferred
+def sendEmailDeferred(emailKey: db.KeyClass):
+	"""
+		Callback from the Taskqueue to send the given Email
+		:param emailKey: Database-Key of the email we should send
+	"""
+	logging.debug("Sending deferred email: %s" % str(emailKey))
+	queueEntity = db.Get(emailKey)
+	assert queueEntity, "Email queue object went missing!"
+	if queueEntity["isSend"]:
+		return True
+	elif queueEntity["errorCount"] > 3:
```

Comment:
Emails are being send through a custom queue (viur-emails). I've added documentation on how it should be configured (so that it retries only once an hour)